app/background.cc

```diff
@@ -143,7 +147,7 @@ std::shared_ptr<background_t> load_background(const std::string& path, int debug
         //  if: can read 2 video frames => it's a video
         //  else: is loaded as an image => it's an image
         //  else: it's not usable.
-        if (pbkd->cap.read(pbkd->raw) && pbkd->cap.read(pbkd->raw)) {
+        if (cnt > -1) {
```

Comment:
If I recall, I chose this current method as `cnt` could be > 1 for some image files (multiple resolutions?) but they would not play as a video... please test with all the variations in `backgrounds` folder :smile:

Reply:
I will perform the tests. Using fps which shall be greater than 0.00000 for a video may be better.

Reply:
Unfortunately, my system don't work with animated.gif. Due to this I can't check if the fps which would be processed on systems allowing this will return a frame rate of 45 Fps (value obtained while converting the file via ffmpeg to a stream.

Jpg file may contain a thumbnail image, do reading twice work, and the file is recognized as video. Reading 3 times will give the expected result, but this shall not be the right way.

The test for picture/video is now based on the fps.  I think that this shall always work.

Reply:
I'll grab your PR and check what I find here in the next day or so - thanks.

Reply:
OK, I've checked the behaviour as the current PR has it (`fps > 0`), and it detects _all_ the backgrounds as videos, but does not fail, since the video loop logic resets the position on each request... we may be able to simplify all this to assume video at all times?

Reply:
I am surprised!
The backsrcrub binary for the cropping branch work as expected, a video is recognized as video and pictures as pictures.

For total_landscaping.jpg I get as output:

```
background properties:
	vid: no
	fcc: 00000000 ()
	fps: 0.000000
	cnt: -1
```
What tell your system? 

I use Fedora 36 XFCE with opencv 4.5.5. What do you use?
May be that we check for fps > 0.001 in order to have the right condition. 

I have tried the following on my development environment:

```
int w1,w2=0;
if(pbkd->cap.read(pbkd->raw)) {
	w1=pbkd->raw.cols;
	if (pbkd->cap.read(pbkd->raw)) {
		w2=pbkd->raw.cols;
	}
}
if(w2 == w1) {
```
This is basically the same as the old condition, but I assume that the width of 2 consecutive images has to be the same.
The final test will pass if we have a video, fail if we have an image, even width thumbnails   (must have a lower size).

The detection was okay on my system.

Reply:
According to opencv cap.get(cv::CAP_PROP_FRAME_COUNT) return the  'Number of frames in the video file'
The first intention to use the count as flag for video or still image, what okay. Asking for CAP_PROP_FPS shall also work.

Work with images as if they were a video is not a great idea. If the image has a big size, reading, ... will take more time as
Retrieving the thumb as now realized within background.cc (Your suggestion).

Reply:
@jjsarton I'm running on Debian stable (11), OpenCV 4.5.1, here's what I get for each background type:
 * `animated.gif` => vid: yes, fps: 45.0.., cnt: 36
 * `background_bauhaus.png` => vid: yes, fps: 25.0.., cnt: -2147483648
 * `rotating_earth.webm` => vid: yes, fps: 30.0.., cnt: 916
 * `total_landscaping.jpg` => vid: yes, fps: 25.0.., cnt: 1

..so this looks like we are heavily dependant on unstable OpenCV behaviour :disappointed: